osbuild/solver/dnf.py

```diff
@@ -110,21 +115,27 @@ def __init__(self, request, persistdir, cache_dir, license_index_path=None):
         self.license_index_path = license_index_path
 
     @staticmethod
-    def _dnfrepo(desc, parent_conf=None):
+    def _dnfrepo(desc, parent_conf=None, root_dir=None):
         """Makes a dnf.repo.Repo out of a JSON repository description"""
 
         repo = dnf.repo.Repo(desc["id"], parent_conf)
+        config = libdnf.conf.ConfigParser
 
         if "name" in desc:
             repo.name = desc["name"]
 
+        def subs(basestr):
+            if not root_dir:
+                return basestr
+            return config.substitute(ucd(basestr), parent_conf.substitutions)
+
         # at least one is required
         if "baseurl" in desc:
-            repo.baseurl = desc["baseurl"]
+            repo.baseurl = [subs(repo) for repo in desc["baseurl"]]
```

Comment:
quick-drive-by: the original code did not had a list, maybe the diff I pasted as a suggestion was wrong (if so, sorry!). This should probably be `repo.baseurl = subs(desc["baseurl"])`, no? (and then the test change is no longer needed). 

If this is an intentional change and  want this to be a list instead of the current single string IMHO it should its own commit with the reasoning/explaination for the change in the commit message.

Reply:
yeah, Josue is right, `baseurl` was always a list.
See also
https://github.com/osbuild/osbuild/blob/a464815ea83c3b2128667442790112099453ea0b/osbuild/solver/dnf.py#L35